backend/benefit/terms/tests/test_api.py

```diff
@@ -0,0 +1,294 @@
+):
+    application.status = status
+    application.save()
+    add_attachments_to_application(
+        request, application
+    )  # so that attachment validation passes
+
+    data = ApplicationSerializer(application).data
+    data["approve_terms"] = {
+        "terms": applicant_terms.pk,
+        "selected_applicant_consents": [
+            obj.pk for obj in applicant_terms.applicant_consents.all()
+        ][1:],
+    }
+    data["status"] = ApplicationStatus.RECEIVED
+
+    response = api_client.put(
+        get_detail_url(application),
+        data,
+    )
+    application.refresh_from_db()
+    assert response.status_code == 400
+    assert "must explicitly select all the applicant consents required" in str(
+        response.data["approve_terms"]["selected_applicant_consents"]
+    )
+
+
+@pytest.mark.parametrize(
+    "status",
+    [
+        ApplicationStatus.DRAFT,
+        ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED,
+    ],
+)
+def test_approve_terms_too_many_consents(
+    api_client, request, application, applicant_terms, status
+):
+    application.status = status
+    application.save()
+    add_attachments_to_application(
+        request, application
+    )  # so that attachment validation passes
+    other_terms = TermsFactory(effective_from=None)
+    data = ApplicationSerializer(application).data
+    consents = [obj.pk for obj in applicant_terms.applicant_consents.all()] + [
+        obj.pk for obj in other_terms.applicant_consents.all()
+    ]
+    data["approve_terms"] = {
+        "terms": applicant_terms.pk,
+        "selected_applicant_consents": consents,
+    }
+    data["status"] = ApplicationStatus.RECEIVED
+
+    response = api_client.put(
+        get_detail_url(application),
+        data,
+    )
+    application.refresh_from_db()
+    assert response.status_code == 400
+    assert "must explicitly select all the applicant consents required" in str(
+        response.data["approve_terms"]["selected_applicant_consents"]
+    )
+
+
+@pytest.mark.parametrize(
+    "from_status, to_status",
+    [
+        (ApplicationStatus.DRAFT, ApplicationStatus.DRAFT),
+        (
+            ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED,
+            ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED,
+        ),
+        (ApplicationStatus.RECEIVED, ApplicationStatus.RECEIVED),
+        (ApplicationStatus.RECEIVED, ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED),
+        (ApplicationStatus.RECEIVED, ApplicationStatus.ACCEPTED),
+    ],
+)
+def test_approve_terms_ignored_when_not_submitting_application(
+    api_client, request, application, applicant_terms, from_status, to_status
+):
+    """
+    When the application is not being submitted, then any terms_approval in the request is ignored
+    """
+    application.status = from_status
+    application.save()
+    add_attachments_to_application(
+        request, application
+    )  # so that attachment validation passes
```

Comment:
I'd suggest we create a fixture `application_with_attachments` to avoid repeating this code

Reply:
Good idea. I'll merge this in case FE devs need this API but I'll modify PR #284 (terms of service) to include application_with_attachments